omod-1.8/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_8/ConceptResource1_8.java

```diff
@@ -256,57 +261,68 @@ public static void setFullySpecifiedName(Concept instance, String name) {
 		ConceptName fullySpecifiedName = new ConceptName(name, Context.getLocale());
 		instance.setFullySpecifiedName(fullySpecifiedName);
 	}
-	
 	/**
-	 * It's needed, because of ConversionException: Don't know how to handle collection class:
-	 * interface java.util.Collection
-	 * 
-	 * if request to update Concept updates ConceptName, adequate resource takes care of it,
-	 * so this method just adds new and removes deleted names.
 	 * 
 	 * @param instance
-	 * @param names
+	 * @param propertyName name of property to set
+	 * @param values objects to be added to property
+	 * @param valueProperties names of properties which will be checked for equality to determine 
+	 * if adding/removing value is needed
+	 * @throws IllegalAccessException
+	 * @throws InvocationTargetException
+	 * @throws NoSuchMethodException
 	 */
-	@PropertySetter("names")
-	public static void setNames(Concept instance, List<ConceptName> names) {
-		for(ConceptName name : names){
-			name.setConcept(instance);
+	protected static void setHibernateCollectionProperty(Concept instance, String propertyName, Collection<?> values, Collection<String> valueProperties) 

```

Comment:
I'd refactor the method to
protected static void <T> modifyCollectionProperty(Concept instance, String propertyName, Collection<T> values,
Comparator<T> valuesComparator)

modifyCollectionProperty(concept, "names", values, new Comparator<ConceptName>() {

});